src/DBmysqlIterator.php

```diff
@@ -139,9 +139,10 @@ public function buildQuery($table, $crit = "", $log = false)
         }
 
         if ($is_legacy) {
-           //if ($_SESSION['glpi_use_mode'] == Session::DEBUG_MODE) {
-           //   trigger_error("Deprecated usage of SQL in DB/request (full query)", E_USER_DEPRECATED);
-           //}
+            Toolbox::deprecated(
+                'Direct query usage calling DBmysqlIterator is strongly discouraged! Use DB::request() instead.',
```

Comment:
```suggestion
                'Direct query usage calling DBmysqlIterator is strongly discouraged! Use DBmysql::doQuery() instead.',
```

Reply:
I do not agree. We should not tell users to use doQuery, but request, in a proper way. In core solution was indeed to use doQuery, but that won't be the case in some plugins (glpiinventory for example.)

Reply:
> Maybe could we just wrote `Direct query usage is strongly discouraged!`

That's what I've applied in last commit

Reply:
Yes that was a """""feature""""" :D